jf_agent/main.py

```diff
@@ -561,19 +550,46 @@ def _s3_cmd(cmd):
 
     print('Sending data to Jellyfish... ')
 
-    output_basedir, output_dir_timestamp = os.path.split(outdir)
-    s3_uri_prefix_with_timestamp = os.path.join(s3_uri_prefix, output_dir_timestamp)
-    done_file_path = f'{os.path.join(outdir, ".done")}'
+    signed_urls = get_signed_url(os.listdir(config.outdir))
+
+    threads = []
+    thread_exceptions = [None] * (len(signed_urls) + 1)
+    for index, file_info in list(enumerate(signed_urls)):
+        # file_info shape {filename: (path_to_obj, signed_url object)}
```

Comment:
See PR on the JF side, but I think the shape of the data here doesn't make sense.  Why would we have a list of single-element dicts?  If this were simply a dict of filename to URL, this whole loop could be something like 

```
threads = [
    threading.Thread(target=upload_file, args=[index, filename, url])
    for index, (filename, url) in enumerate(signed_urls.items())
]

for t in threads:
    t.start()
```